### ISSUE TITLE
`possibly-unresolved-reference` false positive inside function when both variable and function are defined in the same branch of an if statement

### ISSUE THREAD

**Issue by DetachHead** [astral-sh/ty#607]
### Summary

```py
def _(value: bool):
   if value:
      a = 1

      def asdf():
         print(a) # possibly-unresolved-reference
```
https://play.ty.dev/f90a868c-cc44-4d7d-af01-9112b87e384a

### Version

0.0.1-alpha.8 (c1337c962 2025-06-02)

**Comment by AlexWaygood** [comment]
Thanks! This is a duplicate of #210 (see https://github.com/astral-sh/ty/issues/210#issuecomment-2859029566)

**Comment by carljm** [comment]
I suspect the initial version of #210 will not fix this issue; the top-priority fix there is to stop wrongly assuming end-of-scope (and instead assume might-be-called-from-anywhere-in-scope). Once we have that foundation in place, we can explore further trimming of the possibilities.

**Comment by DetachHead** [comment]
In that case should we re-open this issue?

**Comment by carljm** [comment]
> In that case should we re-open this issue?

Yeah I think it's useful to track this as a separate improvement.

**Comment by carljm** [comment]
This was closed by https://github.com/astral-sh/ruff/pull/18750, although it's not fixed.

I think we can leave it closed though, as a duplicate of #710

**Comment by sharkdp** [comment]
I think the boundness analysis problem here was indeed fixed. In a rather crude way, though: by always considering nonlocal symbols to be bound.
